iam/newsletter.py

```diff
@@ -0,0 +1,102 @@
+"""Handle newsletter subscriptions for users."""
+
+from requests import get
+from discord.ext.commands import Cog, command
+from logging import DEBUG, INFO
+
+from iam.db import MemberKey
+from iam.log import new_logger
+from iam.config import PREFIX, NEWSLETTER_SUB_URL
+from iam.hooks import (
+    pre, post, check, log_attempt, log_invoke, log_success,
+    is_strictly_verified_user
+)
+
+LOG = new_logger(__name__)
+"""Logger for this module."""
+
+COG_NAME = "Newsletter"
+"""Name of this module's Cog."""
+
+def setup(bot):
+    """Add Newsletter cog to bot and set up logging.
+
+    Args:
+        bot: Bot object to add cog to.
+    """
+    LOG.debug(f"Setting up {__name__} extension...")
+    cog = Newsletter(bot, LOG)
+    LOG.debug(f"Initialised {COG_NAME} cog")
+    bot.add_cog(cog)
+    LOG.debug(f"Added {COG_NAME} cog to bot")
+
+def teardown(bot):
+    """Remove Newsletter cog from bot and remove logging.
+
+    Args:
+        bot: Bot object to remove cog from.
+    """
+    LOG.debug(f"Tearing down {__name__} extension...")
+    bot.remove_cog(COG_NAME)
+    LOG.debug(f"Removed {COG_NAME} cog from bot")
+    for handler in LOG.handlers:
+        LOG.removeHandler(handler)
+
+async def proc_subscribe(db, user, channel):
+    """Subscribe a user to the newsletter using their stored email.
+
+    Args:
+        db: Database object.
+        user: User object to subscribe.
+        channel: Channel to send confirmation message to.
+    """
+    member_data = db.get_member_data(user.id)
+    zid = member_data[MemberKey.ZID]
+    get(f"{NEWSLETTER_SUB_URL}"
```

Comment:
This does not look sanitised. Please look into libraries for doing correct URL parsing.

Reply:
Module has been rewritten to use the official Mailchimp library